doc/nrf/releases_and_maturity/migration/2.4.99-cs3_to_2.6.99-cs2/migration_guide_2.4.99-cs3_to_2.6.99-cs2_application.rst

```diff
@@ -0,0 +1,142 @@
+     * PPR core: ``nrf54h20dk_nrf54h20_cpuppr`` (Previously ``nrf54h20pdk_nrf54h20_cpuppr@soc1``)
+
+     The previously used SOC1-based board files have been removed.
+
+Security
+========
+
+.. toggle::
+
+   * For samples using ``CONFIG_NRF_SECURITY``:
+
+     * RSA keys are no longer enabled by default.
+       This reduces the code size by 30 kB if not using RSA keys.
+       This also breaks the configuration if using the RSA keys without explicitly enabling an RSA key size.
+       Enable the required key size to fix the configuration, for example by setting the Kconfig option :kconfig:option:`CONFIG_PSA_WANT_RSA_KEY_SIZE_2048` if 2048-bit RSA keys are required.
+
+     * The PSA config is now validated by the :file:`ncs/nrf/ext/oberon/psa/core/library/check_crypto_config.h` file.
+       Users with invalid configurations must update their PSA configuration according to the error messages that the :file:`check_crypto_config.h` file provides.
+
+   * For the :ref:`crypto_persistent_key` sample:
+
+     * The Kconfig option ``CONFIG_PSA_NATIVE_ITS`` is replaced by the Kconfig option :kconfig:option:`CONFIG_TRUSTED_STORAGE`, which enables the new :ref:`trusted_storage_readme` library.
+       The :ref:`trusted_storage_readme` library provides the PSA Internal Trusted Storage (ITS) API for build targets without TF-M.
+       It is not backward compatible with the previous PSA ITS implementation.
+       Migrating from the PSA ITS implementation, enabled by the ``CONFIG_PSA_NATIVE_ITS`` option, to the new :ref:`trusted_storage_readme` library requires manual data migration.
+
+   * For :ref:`lib_wifi_credentials` library and Wi-Fi samples:
+
+     * ``CONFIG_WIFI_CREDENTIALS_BACKEND_PSA_UID_OFFSET`` has been removed because it was specific to the previous solution that used PSA Protected Storage instead of PSA Internal Trusted Storage (ITS).
+       Use :kconfig:option:`CONFIG_WIFI_CREDENTIALS_BACKEND_PSA_OFFSET` to specify the key offset for PSA ITS.
+       Be aware that Wi-Fi credentials stored in Protected Storage will not appear in ITS when switching.
+       To avoid re-provisioning Wi-Fi credentials, manually read out the old credentials from Protected Storage in the previously used UID and store to ITS.
+
+zcbor
+=====
+
+.. toggle::
+
+   * If you have zcbor-generated code that relies on the zcbor libraries through Zephyr, you must regenerate the files using zcbor 0.8.1.
+     Note that the names of generated types and members has been overhauled, so the code using the generated code must likely be changed.
+
+     For example:
+
+      * Leading single underscores and all double underscores are largely gone.
+      * Names sometimes gain suffixes like ``_m`` or ``_l`` for disambiguation.
+      * All enum (choice) names have now gained a ``_c`` suffix, so the enum name no longer matches the corresponding member name exactly (because this broke C++ namespace rules).
```

Comment:
```suggestion
      * All enum (choice) names have now gained a ``_c`` suffix, so the enum name no longer matches the corresponding member name exactly (because this previously broke the C++ namespace rules).
```